pkg/requests/builder.go

```diff
@@ -58,7 +58,7 @@ func (r *builder) WithMethod(method string) Builder {
 
 // WithHeaders replaces the request header map with the given header map.
 func (r *builder) WithHeaders(header http.Header) Builder {
-	r.header = header
+	r.header = header.Clone()
```

Comment:
Why is this change necessary?

Reply:
Ah, this is a remnant of an earlier attempt to add it in the `Builder` interface. In the current code base I don't think it is strictly necessary, but if you combine `WithHeaders` and `SetHeader` you are mutating the original header you passed. I left it because it look strange to me, but I can change it if that is preferred.